komodo/show_version.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+import argparse
+import yaml
+import os
+import re
+from pathlib import Path
+import configparser
+from typing import Union
+from typing import Optional
+
+
+def guess_release() -> str:
+    """
+    Try to sense the Komodo release from the active environment.
+    """
+    try:
+        return os.environ["KOMODO_RELEASE"]
+    except KeyError:
+        raise Exception("Could not sense the name of the release, please provide it.")
+
+
+def read_config(fname: Union[str, Path]) -> dict:
+    """
+    Read an INI file (aka config file) that does not have sections.
+    Creates a temporary 'MAIN' section and pass bask its contents.
+    """
+    config = configparser.ConfigParser()
+    with open(fname, "rt") as f:
+        config.read_string("[MAIN]\n" + f.read())
+    return dict(config["MAIN"])
+
+
+def guess_komodoenv_path(release: str) -> Path:
+    """
+    Use the release name to find the 'real' release path, but from a
+    komodoenv environment. These environments overwrite the environment
+    variable $KOMODO_RELEASE and $PATH, but they store the original path
+    in the komodoenv at 'root/pyvenv.cfg'.
+
+    Args:
+        release (str): The name of the release, e.g. from $KOMODO_RELEASE.
```

Comment:
We don't need to specify the type inside the docstrings, it is (or can be) inferred from the function signature.

Reply:
Fair point, esp since two of them were wrong :grimacing: